### ISSUE TITLE
[Dropdown] When using referenceElement, "aria-controls" and other properties are missing or blank

### ISSUE THREAD

**Issue by mpriour** [Esri/calcite-components#14644]
**monday.com sync:** #12294536273

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Dropdown

### Summary

When using a reference element as the trigger for a dropdown, certain "aria" properties are blank or not present on the trigger. These properties are needed for proper a11y compliance and usability of the component by users using screen readers.
* `aria-controls` is present but blank, but should be the internal div with the role of "menu" or failing that at least the "calcite-dropdown" element.
* `aria-haspopup` is missing, but should be "menu".
* `aria-describedby` is present but blank, but should be the `label` property of the "dropdown" element if there is one.

### Actual Behavior

<img width="1727" height="828" alt="Image" src="https://github.com/user-attachments/assets/98da5927-a523-4245-8788-bc3287023685" />

On the triggerBtn:
* `aria-controls` is present but blank
* `aria-haspopup` is missing
* `aria-describedby` is present but blank

### Expected Behavior

On the triggerBtn
* `aria-controls` should be the internal div with the role of "menu" or failing that at least the "calcite-dropdown" element.
* `aria-haspopup` should be "menu".
* `aria-describedby` should be the `label` property of the "dropdown" element if there is one.

### Reproduction Sample

https://codepen.io/mpriour/pen/azpBjaX?editors=100

### Reproduction Steps

Inspect the `calcite-button` with an `id="triggerBtn"` and observe that the blank or missing `aria-*` attributes.

### Reproduction Version

5.1

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Online

**Comment by isaacbraun** [comment]
@mpriour thanks for logging this! With `referenceElement` and our recent support of multiple reference elements etc., the aria properties used are the `xElements` versions that contain an array of elements and do not show their values in the DOM.

You can access their values through JS, and if you log out the below properties in the CodePen you should see an array with `calcite-dropdown` inside of it.
- `ariaControlsElements` ([MDN ref](https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaControlsElements))
- `ariaDescribedByElements` ([MDN ref](https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaDescribedByElements))

<img width="215" height="95" alt="Image" src="https://github.com/user-attachments/assets/72806ed8-f562-4b6a-b3b2-372394113bec" />

I believe that the attributes still display in the DOM as "empty" to indicate that the property is set, but I may be misunderstood with that. Additionally, I don't think we are setting `ariaHasPopup`. cc @driskull for expertise and thoughts.

**Comment by driskull** [comment]
> Additionally, I don't think we are setting ariaHasPopup. cc @driskull for expertise and thoughts.

We're not currently. If that needs to be done a separate issue would be good.

**Comment by mpriour** [comment]
Thanks @isaacbraun and @driskull, I understand this issue a lot better now. I didn't realize that the referenceElement would have the `*Elements` versions available. Hopefully, both screen readers and a11y scanning tools will catch up to this. I know VoiceOver on Mac and axe tools both had trouble with this when I checked a few weeks ago. However, that's not a calcite problem. We can close this issue and if I get reports about needing the `hasPopup` attribute once we transition to using the `referenceElement` pattern, I'll open a separate, focused issue.

**Comment by isaacbraun** [comment]
Happy to hear that! 😊 That's good context about the current state of a11y tooling, thanks 🤜💥🤛

Closing this out as not planned, but as you mentioned please create a more focused issue in the future if needed 👍
